src/Rule/LatteTemplatesRule.php

```diff
@@ -181,7 +179,7 @@ private function analyseTemplates(array $templates, Scope $scope, array &$errors
                         $includedTemplatePath,
                         $actualClass,
                         $actualAction,
-                        array_merge($collectedTemplateRender->getVariables(), $template->getVariables()),
+                        VariablesHelper::merge($template->getVariables(), $collectedTemplateRender->getVariables()),
```

Comment:
For associative arrays (with variable names as keys) operator `+` and `array_merge` behaves the same (except swaped priorities of conflicting keys). Simple array_merge gives wrong result because variables are list without names as keys. After this change it behaves just like `+` used in compiled code.